param/parameterized.py

```diff
@@ -845,15 +845,6 @@ def __new__(cls_, *args, **kwargs):
             values['precedence'] = 0
         return super().__new__(cls_, **values)
 
-    def __iter__(self):
```

Comment:
@philippjfr are you fine with this removal? I needed to do that to fix pickling as explained in the post, or let's say that was the easier way to fix pickling I found.

Reply:
Hmm, did you try running the Panel test suite with this?

Reply:
Here's the Panel code that I think will break:

```python
        link_args = tuple(cb)
        # Compatibility with Param versions where precedence is dropped
        # from iterator for backward compatibility with older Panel versions
        if 'precedence' in Watcher._fields and len(link_args) < len(Watcher._fields):
            link_args += (cb.precedence,)
        link = LinkWatcher(*(link_args+(target, links, callbacks is not None, bidirectional_watcher)))
        self._links.append(link)
```

Reply:
> Hmm, did you try running the Panel test suite with this?

Yes the tests ran fine with this branch checked out.

That code seemed to deal with backward compatibility with older Panel versions. Given that they pin `param < 2.0`, aren't we safe? With the change made in this PR, `len(link_args) < len(Watcher._fields)` is `False`.

Reply:
Ah, missed that.